Core/Src/internalResistance.c

```diff
@@ -0,0 +1,221 @@
+/* ==================================================================== */
+/* ============================= INCLUDES ============================= */
+/* ==================================================================== */
+
+#include "internalResistance.h"
+#include "math.h"
+
+/* ==================================================================== */
+/* ========================= LOCAL VARIABLES ========================== */
+/* ==================================================================== */
+
+// Discrete buffers hold consecutive sensor readings from the bms
+static float currentDiscreteBuffer[DISCRETE_CURRENT_BUFFER_SIZE];
+static float voltageDiscreteBuffer[NUM_BMBS_PER_PACK][NUM_BRICKS_PER_BMB][DISCRETE_VOLTAGE_BUFFER_SIZE];
+
+// Discrete buffer pointers hold current index of discrete buffers
+static uint32_t discreteCurrentBufferIndex = 0;
+static uint32_t discreteVoltageBufferIndex = 0;
+
+// Data good indicates that all sensor readings in discrete buffers are good
+static bool currentDataGood = true;
+static bool voltageDataGood[NUM_BMBS_PER_PACK][NUM_BRICKS_PER_BMB] = {true};
```

Comment:
Does the initialization work this way? I thought this only guarantees that the first value is initialized?